content/getting-started/ruby.textile

```diff
@@ -0,0 +1,220 @@
+
+* Create a @get_started.rb@ file in your project and add the following function to instantiate the SDK and establish a connection to Ably. At the minimum you need to provide an authentication mechanism. Use an API key for simplicity, but you should use token authentication in a production app. A @client_id@ ensures the client is identified, which is required to use certain features, such as presence:
+
+```[ruby]
+require 'ably'
+require 'eventmachine'
+
+def get_started
+  EventMachine.run do
+    # Initialize the Ably Realtime client
+    realtime_client = Ably::Realtime.new(key: '{{API_KEY}}', client_id: 'my-first-client')
+
+    # Wait for the connection to be established
+    realtime_client.connection.on(:connected) do
+      puts "Made my first connection!"
+    end
+
+
+    realtime_client.connection.once(:connected) do
+
+    end
+  end
+end
+
+get_started
+```
+
+You can monitor the lifecycle of clients' connections by registering a listener that will emit an event every time the connection state changes. For now, run the function with @ruby get_started.rb@ to log a message to the console to know that the connection attempt was successful. You'll see the message printed to your console, and you can also inspect the connection event in the dev console of your app.
+
+h2(#step-2). Step 2: Subscribe to a channel and publish a message
+
+Messages contain the data that a client is communicating, such as a short 'hello' from a colleague, or a financial update being broadcast to subscribers from a server. Ably uses channels to separate messages into different topics, so that clients only ever receive messages on the channels they are subscribed to.
+
+* Add the following lines to your @get_started@ method within the @EventMachine@ to create a channel instance and register a listener to subscribe to the channel. Then run it with @ruby get_started.rb@:
+
+```[ruby]
+# Get a channel instance
+channel = realtime_client.channels.get('my-first-channel')
+
+# Subscribe to messages on the channel
+channel.subscribe do |message|
+  puts "Received message: #{message.data}"
+end
+```
+
+* Use the Ably CLI to publish a message to your first channel. The message will be received by the client you've subscribed to the channel, and be logged to the console.
+
+```[sh]
+ably channels publish my-first-channel 'Hello!'
+```
+
+* In a new terminal tab, subscribe to the same channel using the CLI:
+
+```[sh]
+ably channels subscribe my-first-channel
+```
+
+Publish another message using the CLI and you will see that it's received instantly by the client you have running locally, as well as the subscribed terminal instance.
+
+To publish a message in your code, you can add the following line to your @get_started@ method after subscribing to the channel:
+
+```[ruby]
+channel.publish 'example', 'A message sent from my first client!'
```

Comment:
I'm not sure it's clear to the user why there are two arguments here, especially given that the CLI `publish` call above only took one argument?

Reply:
I've added `name` to the CLI, data is specified there by default. So to keep things consistent I've added the name flag.